cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -335,8 +337,8 @@ public StreamConfig call() throws Exception {
         alterExploreStream(streamId, true, config.getFormat());
         streamMetaStore.addStream(streamId);
         publishAudit(streamId, AuditType.CREATE);
-        AbstractSystemMetadataWriter systemMetadataWriter = new StreamSystemMetadataWriter(metadataStore, streamId,
-                                                                                           config);
+        AbstractSystemMetadataWriter systemMetadataWriter =
```

Comment:
We have class named `Abstract*` as the root class? That's not a very good practice. By convention, you usually have an interface to define the behavior/contract and the `Abstract` class is to provide common/default behaviors as well governing the control flow.